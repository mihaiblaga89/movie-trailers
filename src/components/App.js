import React from 'react';
import { Container } from 'semantic-ui-react';
import { Router } from '@reach/router';

import Home from './pages/Home';
import Detail from './pages/Detail';

const NotFound = () => <div>Page not found</div>;

const App = () => (
    <Container style={{ marginTop: '50px' }}>
        <Router>
            <Home path="/" />
            <Detail path="movies/:movieId" />
            <Detail path="series/:seriesId" />
            <NotFound path="*" />
        </Router>
    </Container>
);

export default App;
