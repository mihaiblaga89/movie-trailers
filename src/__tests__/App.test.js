import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from '../components/App';

/* global document */

it('renders without crashing and matches snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
