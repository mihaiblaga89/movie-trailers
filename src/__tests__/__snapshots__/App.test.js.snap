// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <div
    style={
      Object {
        "outline": "none",
      }
    }
    tabIndex="-1"
  >
    <h1
      className="ui header"
    >
      Movie Trailers
    </h1>
    <div
      className="ui fluid search search-field"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ui icon input"
      >
        <input
          autoComplete="off"
          className="prompt"
          onChange={[Function]}
          onClick={[Function]}
          tabIndex="0"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          className="search icon"
          onClick={[Function]}
        />
      </div>
      <div
        className="results transition"
      >
        <div
          className="message empty"
        >
          <div
            className="header"
          >
            No results found.
          </div>
        </div>
      </div>
    </div>
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <h2
        className="ui header"
      >
        Popular movies
      </h2>
      <div
        className="ui fluid placeholder"
        style={
          Object {
            "height": "200px",
          }
        }
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <h2
        className="ui header"
      >
        Popular series
      </h2>
      <div
        className="ui fluid placeholder"
        style={
          Object {
            "height": "200px",
          }
        }
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <h2
        className="ui header"
      >
        Family
      </h2>
      <div
        className="ui fluid placeholder"
        style={
          Object {
            "height": "200px",
          }
        }
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <h2
        className="ui header"
      >
        Documentary
      </h2>
      <div
        className="ui fluid placeholder"
        style={
          Object {
            "height": "200px",
          }
        }
      >
        <div
          className="rectangular image"
        />
      </div>
    </div>
  </div>
</div>
`;
